File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        String strx = Integer.toString(x);
        int i = 0;
        int j = strx.length() - 1;

        while(i<=j){
            if(strx.charAt(i) == strx.charAt(j)){
                i++;
                j--;
            }
            else{
                return false;
            }
        }
        return true;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        num_str = str(x)

        i, j = 0, len(num_str) - 1

        while i<=j:
            if num_str[i] != num_str[j]:
                return False
            i += 1
            j -= 1
        
        return True